src/web/session-summary.jsp

```diff
@@ -94,100 +94,72 @@
+    List<ClientSession> sessions = sessionManager.getSessions(sessionResultFilter)
         .stream()
-        .filter(new Predicate<ClientSession>() {
-            @Override
-            public boolean test(final ClientSession clientSession) {
-                try { // skip invalid sessions (OF-590)
-                    return clientSession.validate();
-                } catch (Exception ex) {
-                    return false;
-                }
+        .filter(clientSession -> {
+            try { // skip invalid sessions (OF-590)
+                return clientSession.validate();
+            } catch (Exception ex) {
+                return false;
             }
         })
         .collect(Collectors.toList());
 
     // By default, display all nodes
-    Predicate<ClientSession> filter = new Predicate<ClientSession>() {
-        @Override
-        public boolean test(final ClientSession session) {
-            return true;
-        }
-    };
+    Predicate<ClientSession> filter = clientSession -> true;
     final String searchName = ParamUtils.getStringParameter(request, "searchName", "");
     if(!searchName.trim().isEmpty()) {
         final String searchCriteria = searchName.trim();
-        filter = filter.and(new Predicate<ClientSession>() {
-            @Override
-            public boolean test(final ClientSession session) {
-                return StringUtils.containsIgnoringCase(session.getAddress().getNode(), searchCriteria);
-            }
-        });
+        filter = filter.and(clientSession -> StringUtils.containsIgnoringCase(clientSession.getAddress().getNode(), searchCriteria));
     }
     final String searchResource = ParamUtils.getStringParameter(request, "searchResource", "");
     if(!searchResource.trim().isEmpty()) {
         final String searchCriteria = searchResource.trim();
-        filter = filter.and(new Predicate<ClientSession>() {
-            @Override
-            public boolean test(final ClientSession session) {
-                return StringUtils.containsIgnoringCase(session.getAddress().getResource(), searchCriteria);
-            }
-        });
+        filter = filter.and(clientSession -> StringUtils.containsIgnoringCase(clientSession.getAddress().getResource(), searchCriteria));
     }
     final String searchNode = ParamUtils.getStringParameter(request, "searchNode", "");
-    if(!searchNode.isEmpty()) {
-        filter = filter.and(new Predicate<ClientSession>() {
-            @Override
-            public boolean test(final ClientSession session) {
-                return searchNode.equals("local") && session instanceof LocalClientSession;
-            }
-        });
+    if(searchNode.equals("local")) {
+        filter = filter.and(LocalClientSession.class::isInstance);
+    } else if (searchNode.equals("remote")) {
+        filter = filter.and(clientSession -> !LocalClientSession.class.isInstance(clientSession));
```

Comment:
This doesn't appear to be a one-on-one conversion from Predicate to Lambda. Is that intentional?

Reply:
Yes, it is. The same effect in the end, but a lot clearer as to the intent (IMHO)